File: VIF_git/GSDS-Qual/LeetCode/week1/Linked_List/Reverse_Linked_List/sunghyun.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        currptr = head
        prev = None
        
        while(currptr != None):
            nextptr = currptr.next
            currptr.next = prev
            prev = currptr
            if(nextptr == None):
                break
            currptr = nextptr
        
        return currptr
        

File: VIF_git/GSDS-Qual/LeetCode/week1/Linked_List/Reverse_Linked_List/sunghyun.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* reverseList(struct ListNode* head) {
    struct ListNode* currptr = head;
    struct ListNode* prev = NULL;

    while(currptr != NULL){
        struct ListNode* nextptr = currptr->next;
        currptr->next = prev;
        prev = currptr;
        if(nextptr == NULL){
            break;
        }
        currptr = nextptr;
    }

    return currptr;
}